Modules/Server/Sources/PocketCastsServer/Public/Sync/UpNextSyncTask.swift

```diff
@@ -50,34 +50,51 @@ class UpNextSyncTask: ApiBaseTask {
         var upNextChanges = Api_UpNextChanges()
         var latestActionTime: Int64 = 0
         var changes = [Api_UpNextChanges.Change]()
-        // replace action
-        if let replaceAction = DataManager.sharedManager.findReplaceAction() {
-            if replaceAction.utcTime > latestActionTime {
-                latestActionTime = replaceAction.utcTime
-            }
-            if let action = convertToProto(action: replaceAction) {
-                changes.append(action)
-            }
-        }
 
-        // all other add/remove actions
-        let actions = DataManager.sharedManager.findUpdateActions()
-        if actions.count > 0 {
-            for action in actions {
-                if action.utcTime > latestActionTime {
-                    latestActionTime = action.utcTime
+        FileLog.shared.addMessage("UpNextSyncTask [createUpNextSyncRequest]: Sync Reason? \(ServerSettings.syncReason?.rawValue ?? "None")")
+
+        // The process for syncing the up next queue involves sending all local changes to the server, which then attempts
+        // to apply those changes to the stored queue in the database. Once complete, the modified queue is saved in the
+        // database and sent back to us. The changes are then applied to the local queue by the "applyServerChanges" logic.
+        //
+        // This works well when each device keeps up to date and tracks its own changes to the queue, but can be problematic if a user
+        // signs into an existing account using a device whose queue is completely different from what is stored on the server,
+        // such as a user who installs the app, adds items to their queue, and then signs into an existing account. This
+        // can result in the queue being completely replaced by the new devices changes.
+        //
+        // To prevent this, during a login we no longer send any local changes to the server, instead we pull the latest sync'd
+        // up next queue and attempt to merge the changes later in the "applyServerChanges" call.
```

Comment:
> ...we no longer send any local changes to the server, instead we pull the latest sync'd up next queue and attempt to merge the changes

FWIW, it looks Android already wasn't sending the local changes to the server until after it merged the queues, so this aligns well (Android's problem was that it was "merging" the server and local queues [by removing any locally queued episodes that weren't in the server queue](https://github.com/Automattic/pocket-casts-android/blob/d1502d9bf89fec3e19098f3ec78bc41e1473b447/modules/services/model/src/main/java/au/com/shiftyjelly/pocketcasts/models/db/dao/UpNextDao.kt#L182-L185) 😬 ).